$OpenBSD: patch-src_kit_kit-spawn_c,v 1.1 2009/10/15 21:51:54 ajacoutot Exp $

From FreeBSD:
Fix a problem where waitpid() may return EINTR which causes the various
polkit helpers to report a failure.  This fix allows dynamic polkit
policies to actually work (e.g. mounting fixed drives with proper
authorization).

--- src/kit/kit-spawn.c.orig	Fri May 30 23:24:44 2008
+++ src/kit/kit-spawn.c	Thu Oct 15 23:41:44 2009
@@ -396,7 +396,10 @@ kit_spawn_sync (const char     *working_directory,
                         }
                 }
 
+again:
                 if (waitpid (pid, out_exit_status, 0) == -1) {
+			if (errno == EINTR)
+				goto again;
                         goto out;
                 }
                 pid = -1;
